### PR TITLE
[Windows] Add MySQL to the toolset

### PR DESCRIPTION
**Pull request by V-Zabayrachny**
# Description
This pull request moves the MySQL versions to the toolset. Also added a check to the pester tests to make sure we are installing the correct version with accordance to the toolset.

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/toolsets/toolset-2022.json [hunk 1]
```diff
@@ -304,5 +304,9 @@
     },
     "node": {
         "default": "14.18.1"
+    },
+    "Mysql": {
+        "short_version": "8",
+        "full_version": "8.0.26"
```
Comment (miketimofeev) [inline images/win/toolsets/toolset-2022.json:310]: Is it possible to download the latest 8.x.x version instead of hardcoding the exact one?

Comment (V-Zabayrachny) [reply images/win/toolsets/toolset-2022.json]: Yes, I made changes for this

File: images/win/toolsets/toolset-2019.json [hunk 2]
```diff
@@ -456,7 +456,6 @@
         "default": "14.18.1"
     },
     "Mysql": {
-        "short_version": "5.7",
-        "full_version": "5.7.35"
+        "short_version": "5.7"
```
Comment (miketimofeev) [inline images/win/toolsets/toolset-2019.json:459]: Just version
```suggestion
        "version": "5.7"
```

File: images/win/scripts/Installers/Install-MysqlCli.ps1 [hunk 3]
```diff
@@ -11,13 +11,9 @@ $ArgumentList = ("/install", "/quiet", "/norestart")
...
 
 ## Downloading mysql
-if (Test-IsWin22) {
-    $MysqlVersionName = "mysql-8.0.26-winx64"
-    $MysqlVersionUrl = "https://dev.mysql.com/get/Downloads/MySQL-8/${MysqlVersionName}.zip"
-} else {
-    $MysqlVersionName = "mysql-5.7.35-winx64"
-    $MysqlVersionUrl = "https://dev.mysql.com/get/Downloads/MySQL-5.7/${MysqlVersionName}.zip"
-}
+$MysqlShortVersion = (Get-ToolsetContent).Mysql.short_version
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-MysqlCli.ps1:14]:
```suggestion
$MysqlMajorMinor = (Get-ToolsetContent).Mysql.version
```

File: images/win/scripts/Installers/Install-MysqlCli.ps1 [hunk 4]
```diff
@@ -11,13 +11,9 @@ $ArgumentList = ("/install", "/quiet", "/norestart")
...
+$MysqlFullVersion = ((Invoke-WebRequest -Uri https://dev.mysql.com/downloads/mysql/${MysqlShortVersion}.html).Content | Select-String -Pattern "${MysqlShortVersion}\.\d+").Matches.Value
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-MysqlCli.ps1:15]:
```suggestion
$MysqlFullVersion = ((Invoke-WebRequest -Uri https://dev.mysql.com/downloads/mysql/${MysqlShortVersion}.html).Content | Select-String -Pattern "${MysqlMajorMinor}\.\d+").Matches.Value
```

File: images/win/scripts/Installers/Install-MysqlCli.ps1 [hunk 5]
```diff
@@ -11,13 +11,9 @@ $ArgumentList = ("/install", "/quiet", "/norestart")
...
+$MysqlVersionUrl = "https://dev.mysql.com/get/Downloads/MySQL-${MysqlShortVersion}/mysql-${MysqlFullVersion}-winx64.zip"
```
Comment (miketimofeev) [inline images/win/scripts/Installers/Install-MysqlCli.ps1:16]:
```suggestion
$MysqlVersionUrl = "https://dev.mysql.com/get/Downloads/MySQL-${MysqlMajorMinor}/mysql-${MysqlFullVersion}-winx64.zip"
```

File: images/win/scripts/Tests/Databases.Tests.ps1 [hunk 6]
```diff
@@ -53,8 +53,10 @@ Describe "PostgreSQL" {
...
 Describe "MySQL" {
-    It "MySQL CLI" {
-        "mysql -V" | Should -ReturnZeroExitCode
-    }
-}
+    It "MySQL version should correspond to the version in the toolset" {
+        $MysqlShortVersion = (Get-ToolsetContent).Mysql.short_version
+        $MysqlFullVersion = ((Invoke-WebRequest -Uri https://dev.mysql.com/downloads/mysql/${MysqlShortVersion}.html).Content | Select-String -Pattern "${MysqlShortVersion}\.\d+").Matches.Value
 
+        mysql --version | Should -BeLike "*${MysqlFullVersion}*"
```
Comment (miketimofeev) [inline images/win/scripts/Tests/Databases.Tests.ps1:60]: I don't think that we need to check the exact version here. It is enough to check whether `mysql --version` is like `MysqlMajorMinor` or not.

File: images/win/scripts/Tests/Databases.Tests.ps1 [hunk 7]
```diff
@@ -53,8 +53,10 @@ Describe "PostgreSQL" {
 }
 
 Describe "MySQL" {
-    It "MySQL CLI" {
-        "mysql -V" | Should -ReturnZeroExitCode
-    }
-}
+    It "MySQL version should correspond to the version in the toolset" {
+        $MysqlShortVersion = (Get-ToolsetContent).Mysql.short_version
```
Comment (miketimofeev) [inline images/win/scripts/Tests/Databases.Tests.ps1:57]:
```suggestion
        $MysqlMajorMinor = (Get-ToolsetContent).Mysql.short_version
```

File: images/win/scripts/Tests/Databases.Tests.ps1 [hunk 8]
```diff
@@ -53,8 +53,10 @@ Describe "PostgreSQL" {
 }
 
 Describe "MySQL" {
-    It "MySQL CLI" {
```
Comment (miketimofeev) [inline images/win/scripts/Tests/Databases.Tests.ps1:56]: Why did you remove this test?

Comment (V-Zabayrachny) [reply images/win/scripts/Tests/Databases.Tests.ps1:56]: I fixed it. Thanks.

File: images/win/scripts/Tests/Databases.Tests.ps1 (hunk 7 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Tests/Databases.Tests.ps1]: I fixed it. Thanks.

File: images/win/scripts/Tests/Databases.Tests.ps1 (hunk 6 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Tests/Databases.Tests.ps1]: I fixed it. Thanks.

File: images/win/scripts/Installers/Install-MysqlCli.ps1 (hunk 5 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Installers/Install-MysqlCli.ps1]: I fixed it. Thanks.

File: images/win/toolsets/toolset-2019.json (hunk 2 above)
Comment (V-Zabayrachny) [reply images/win/toolsets/toolset-2019.json]: I fixed it. Thanks.

File: images/win/scripts/Installers/Install-MysqlCli.ps1 (hunk 3 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Installers/Install-MysqlCli.ps1]: I fixed it. Thanks.

File: images/win/scripts/Installers/Install-MysqlCli.ps1 (hunk 4 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Installers/Install-MysqlCli.ps1]: I fixed it. Thanks.

File: images/win/scripts/Tests/Databases.Tests.ps1 [hunk 9]
```diff
@@ -54,7 +54,9 @@ Describe "PostgreSQL" {
 
 Describe "MySQL" {
     It "MySQL CLI" {
-        "mysql -V" | Should -ReturnZeroExitCode
-    }
-}
+        $MysqlMajorMinor = (Get-ToolsetContent).Mysql.version
+        $MysqlFullVersion = ((Invoke-WebRequest -Uri https://dev.mysql.com/downloads/mysql/${MysqlMajorMinor}.html).Content | Select-String -Pattern "${MysqlMajorMinor}\.\d+").Matches.Value
```
Comment (al-cheb) [inline images/win/scripts/Tests/Databases.Tests.ps1:58]: Why do we need this statement?

File: images/win/scripts/Tests/Databases.Tests.ps1 (hunk 6 above)
Comment (miketimofeev) [reply images/win/scripts/Tests/Databases.Tests.ps1]: Looks like no

File: images/win/scripts/Tests/Databases.Tests.ps1 (hunk 9 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Tests/Databases.Tests.ps1]: Deleted it. Thanks.

File: images/win/scripts/Tests/Databases.Tests.ps1 (hunk 6 above)
Comment (V-Zabayrachny) [reply images/win/scripts/Tests/Databases.Tests.ps1]: Deleted it. Thanks.

**Review by miketimofeev — APPROVED** [review]

**Review by mikhailkoliada — DISMISSED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]
